tardis/io/util.py

```diff
@@ -87,6 +87,16 @@ def construct_quantity(self, node):
         return quantity_from_str(data)
 
     def mapping_constructor(self, node):
+        mapping_dict = self.construct_mapping(node)
+        if set(mapping_dict.keys()) == {'start', 'stop', 'num'}:
+            if type(mapping_dict['start']) == int:
```

Comment:
The check here is flawed. What if it is a float? It would be better to check if it is a quantity with `hasattr('unit')` instead of checking if it is a number.